06-regression.Rmd

```diff
@@ -203,66 +207,113 @@ ggplot(data = values, mapping = aes(V1, V2)) +
+1. Recall our earlier computation of the correlation coefficient, which describes the strength of the linear relationship between two numerical variables. Looking at Figure \@ref(fig:numxplot2), it is not immediately apparent that these two variables are positively related. This is to be expected given the positive, but rather weak (close to 0), correlation coefficient of `r cor(evals_small$score, evals_small$bty_avg) %>% round(3)`.
+
+Before we continue, we bring to light an important fact about this dataset: it suffers from *overplotting*. Recall from the data visualization Subsection \@ref(overplotting) that overplotting occurs when several points are stacked directly on top of each other thereby obscuring the number of points. For example, let's focus on the 6 points in the top-right of the plot with a beauty score of around 8 out of 10: are there truly only 6 points, or are there many more just stacked on top of each other? You can think of these as *ties*. Let's break up these ties with a little random "jitter" added to the points in Figure \@ref(fig:numxplot2). 
+
+```{r numxplot2, echo=FALSE, warning=FALSE, fig.cap="Instructor evaluation scores at UT Austin: Jittered"}
 set.seed(76)
+ggplot(evals_small, aes(x = bty_avg, y = score)) +
+  geom_jitter() +
+  labs(x = "Beauty Score", y = "Teaching Score", 
+       title = "Relationship of teaching and beauty scores")
 ```
-```{r numxplot2, warning=FALSE, fig.cap="Instructor evaluation scores at UT Austin: Jittered"}
-ggplot(evals, aes(x = bty_avg, y = score)) +
+
+Jittering adds a little random bump to each of the points to break up these ties: just enough so you can distinguish them, but not so much that the plot is overly altered. Furthemore, jittering is strictly a visualization tool; it does not alter the original values in the dataset. 
+
+Let's compare side-by-side the regular scatterplot in Figure \@ref(fig:numxplot1) with the jittered scatterplot in Figure \@ref(fig:numxplot2) in Figure \@ref(fig:numxplot2-a).
+
+```{r numxplot2-a, warning=FALSE, fig.cap="Comparing regular and jittered scatterplots."}
+box <- data_frame(x=c(7.6, 8, 8, 7.6, 7.6), y=c(4.75, 4.75, 5.1, 5.1, 4.75))
+p1 <- ggplot(evals_small, aes(x = bty_avg, y = score)) +
+  geom_point() +
+  labs(x = "Beauty Score", y = "Teaching Score", 
+       title = "Regular scatterplot") +
+  geom_path(data = box, aes(x=x, y=y), col = "orange", size = 1)
+set.seed(76)
+p2 <- ggplot(evals_small, aes(x = bty_avg, y = score)) +
   geom_jitter() +
-  labs(x = "Beauty Score", y = "Teaching Score", title = "Relationship of teaching and beauty scores")
+  labs(x = "Beauty Score", y = "Teaching Score", 
+       title = "Jittered scatterplot") +
+  geom_path(data = box, aes(x=x, y=y), col = "orange", size = 1)
+p1 + p2
 ```
 
-From Figure \@ref(fig:numxplot2) we make several observations:
+We make several further observations:
 
 <!-- We might want to actually highlight these points in the plot. -->
```

Comment:
We could use {gghighlight} to do so I think.

Reply:
For a one time highlighting of 6 points, I feel like that's overkill. But I would not object you switched this out.